pandas/core/dtypes/dtypes.py

```diff
@@ -817,6 +848,10 @@ def __eq__(self, other):
 
         return isinstance(other, PeriodDtype) and self.freq == other.freq
 
+    def __setstate__(self, state):
+        # for pickle compat.
+        self._freq = state['freq']
+
```

Comment:
do we have a round-trip on this test? shouldn't you also need ``__getstate__``? or maybe not as you are setting _freq which is now good.

Reply:
`__getstate__` is implemented in the super class:

https://github.com/pandas-dev/pandas/blob/455a2cdc4808890dd7eea41494100f48f34794d5/pandas/core/dtypes/dtypes.py#L152-L154

Added in https://github.com/pandas-dev/pandas/commit/154a6478eec50459ae46d28e4393b345c9ab9cd1

Reply:
And it's the round trip test that was failing

Reply:
hmm, what if you move ``_freq`` into metdata then? (and so on for the other subclasses)

Reply:
Yeah, I have been thinking about that as well. `_metadata` is also used for other things though (that are included in the EA interface, this `__getstate__` is only for our internal dtypes). Eg for equality. For equality it would not matter if `freq` or `_freq` is included, but if we would want to use it eg also for a default repr, then it need to be the public attributes

Reply:
hmm, yeah they are coupled. i don't really like having to be explicit about the pickle compat here.....

Reply:
>  i don't really like having to be explicit about the pickle compat here.....

Yep, I agree. Although it is consistent with how it is already done in the other dtypes (Categorical actually has the same), it would be nice to clean it up. I could also add a different `_metadata_pickles` list, so we can handle both getstate and setstate in the PandasExtensionDtype superclass. Of course that doesn't change the implementation, but it does reduce some duplication in the code.